### ISSUE TITLE
Terraform gets only first VM with _number_of_instances or _cluster > 1

### ISSUE THREAD

**Issue by GMZwinge** [vmware-archive/terraform-provider-vra7#39]
**vRA 7.x version**
Version: 7.5.0 (Build: 10053500)
**Terraform version**
Terraform v0.12.16
**terraform-provider-vra7 plugin version**
provider.vra7 v0.5.0

**Describe the bug**
With _number_of_instances > 1 in deployment_configuration or _cluster > 1 in resource_configuration, vRealize provision the number of VM specified in _number_of_instances  or _cluster under a single deployment. But Terraform gets back only the first VM in the deployment.

**To Reproduce**
1. Terraform config file
```
resource "vra7_deployment" "test_deployment" {
  catalog_item_name = "<catalogName>"
  deployment_configuration = {
    # Same with this instead of _cluster below.
    #_number_of_instances = "2"
  }
  resource_configuration = {
    "vSphere__vCenter__Machine_1._cluster" = "2"
    "vSphere__vCenter__Machine_1.name" = ""
  }
}
output "resource_configuration" {
  value = vra7_deployment.test_deployment[*].resource_configuration
}
```
2. Terraform command
terraform init
terraform apply
3. Error
The file terraform.tfstate contains info only on the first VM.
The output lists info only for the first VM:
```
resource_configuration = [
  {
    "vSphere__vCenter__Machine_1._cluster" = "2"
    "vSphere__vCenter__Machine_1.name" = "<machine1Name>"
  },
]
```

**Expected behavior**
- terraform.tfstate file should contain info on all the VMs in the deployment.
- Output should list info for all the VMs in the deployment. Eg:
```
resource_configuration = [
  {
    "vSphere__vCenter__Machine_1._cluster" = "2"
    "vSphere__vCenter__Machine_1.name" = "<machine1Name>"
  },
  {
    "vSphere__vCenter__Machine_1._cluster" = "2"
    "vSphere__vCenter__Machine_1.name" = "<machine2Name>"
  },
]
```

**Screenshots**

**Logs**
Attach logs to help debug the issue
1. Attached [vra-terraform-scrubed.log](https://github.com/terraform-providers/terraform-provider-vra7/files/3880764/vra-terraform-scrubed.log).
2. Attached Terraform console log enabled for trace [console-scrubed.log](https://github.com/terraform-providers/terraform-provider-vra7/files/3880766/console-scrubed.log) (Enable terraform logs following the steps mentioned in https://www.terraform.io/docs/configuration/environment-variables.html)
The vra-terraform.log and the Terraform console log both correctly show the name and IP address for all the VMs, not just the first one. Search for `The resource data map of the resource vSphere__vCenter__Machine_1 is`.

**Desktop (please complete the following information):**
 - OS: Windows 10, version 1703

**Comment by Prativa20** [comment]
@GMZwinge Created a PR as an enhancement that will take care of this
https://github.com/terraform-providers/terraform-provider-vra7/issues/41

Please provide your feedback

**Comment by Prativa20** [comment]
@GMZwinge  This issue has been fixed in the latest version, v1.0.0. 
"cluster" has been added as property under resource_configuration in the schema to handle this and also support scaling.

Please update your resource configuration block.

resource_configuration  {
     component_name = "vSphere__vCenter__Machine_1"
     cluster = 2
     configuration = {
        // properties that needs to be modified
       // name property need not be added here, the state file have that information  
     }
  }

Refer to the README, examples and documentation.
